c90bd8569f20c992f7cd79ca7afc64b5a3b4858f
src/main/java/com/yelp/nrtsearch/server/luceneserver/highlights/NRTFastVectorHighlighter.java
--- 
+++ 
@@ -90,6 +90,9 @@
       fragListBuilder = SINGLE_FRAG_LIST_BUILDER;
       numberOfFragments = Integer.MAX_VALUE;
       fragmentCharSize = Integer.MAX_VALUE;
+    } else if (fragmentCharSize == 0 || fragmentCharSize == Integer.MAX_VALUE) {
+      fragListBuilder = SINGLE_FRAG_LIST_BUILDER;
+      fragmentCharSize = Integer.MAX_VALUE;
     } else {
       fragListBuilder = SIMPLE_FRAG_LIST_BUILDER;
     }